spring-cql/src/test/java/org/springframework/cassandra/test/integration/AbstractEmbeddedCassandraIntegrationTest.java

```diff
@@ -15,81 +15,72 @@
  */
 package org.springframework.cassandra.test.integration;
 
-import java.io.IOException;
 import java.util.UUID;
 
-import org.apache.cassandra.exceptions.ConfigurationException;
-import org.apache.thrift.transport.TTransportException;
-import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
-import org.junit.BeforeClass;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.cassandra.test.integration.support.SpringCqlBuildProperties;
-import org.springframework.cassandra.test.unit.support.Utils;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.springframework.cassandra.core.SessionCallback;
+import org.springframework.cassandra.test.integration.support.CassandraConnectionProperties;
+import org.springframework.dao.DataAccessException;
 
 import com.datastax.driver.core.Cluster;
 import com.datastax.driver.core.Session;
 
 /**
- * Abstract base integration test class that starts an embedded Cassandra instance.
- * 
+ * Abstract base integration test class that starts an embedded Cassandra instance. Test clients can use the
+ * {@link #cluster} instance to create sessions and get access. Expect the {@link #cluster} instance to be closed once
+ * the test has been run.
+ * <p>
+ * This class is intended to be subclassed by integration test classes.
+ *
  * @author Matthew T. Adams
+ * @author Mark Paluch
  */
-public class AbstractEmbeddedCassandraIntegrationTest {
-
-	public static String uuid() {
-		return UUID.randomUUID().toString();
-	}
-
-	static Logger log = LoggerFactory.getLogger(AbstractEmbeddedCassandraIntegrationTest.class);
-
-	protected static String CASSANDRA_CONFIG = "spring-cassandra.yaml";
-	protected static String CASSANDRA_HOST = "localhost";
-
-	protected static SpringCqlBuildProperties PROPS = new SpringCqlBuildProperties();
-	protected static int CASSANDRA_NATIVE_PORT = PROPS.getCassandraPort();
+public abstract class AbstractEmbeddedCassandraIntegrationTest {
 
 	/**
-	 * The session connected to the system keyspace.
+	 * Initiate a Cassandra environment in test class scope.
 	 */
-	protected static Session system;
+	@ClassRule public final static CassandraRule cassandraEnvironment = new CassandraRule("embedded-cassandra.yaml");
 
 	/**
-	 * The {@link Cluster} that's connected to Cassandra.
+	 * Initiate a Cassandra environment in test scope.
 	 */
-	protected static Cluster cluster;
-
-	public static String randomKeyspaceName() {
-		return Utils.randomKeyspaceName();
-	}
+	@Rule public final CassandraRule cassandraRule = cassandraEnvironment.testInstance()
```

Comment:
Assuming the `InvocationMode` is analogous, or similar to `@BeforeClass` and `@Before` annotations in JUnit, indicating whether a setup operation is executed before each test case, or before the entire test suite, we probably only need to assign the test instances `cluster` variable once.  This version of `before` calls for `each`, as the default (?).